crates/sui-data-ingestion-core/src/worker_pool.rs

```diff
@@ -114,6 +114,7 @@ impl<W: Worker + 'static> WorkerPool<W> {
                     if sequence_number < current_checkpoint_number {
                         continue;
                     }
+                    self.worker.preprocess_hook(checkpoint.clone()).expect("failed to preprocess task");
```

Comment:
are checkpoints from `checkpoint_receiver.recv()` following asc order? iirc download are done in parallel and thus downloaded checkpoints can be out of order, and as a result if a package is published on cp X and then the related objects are updated in X+1, but here it's possible that checkpoint X+1 is assigned a worker before package of cp X is pre-processed, which can be problematic?

Reply:
here all checkpoints are guaranteed to be received in order. Download is done in parallel but results are accumulated before sent to the executor/worker pool

Reply:
I see, saw that in reader.rs, that's great!